addons/website_crm_partner_assign/models/crm_lead.py

```diff
@@ -317,3 +317,15 @@ def _get_access_action(self, access_uid=None, force_website=False):
                     'url': '/my/opportunity/%s' % record.id,
                 }
         return super(CrmLead, self)._get_access_action(access_uid=access_uid, force_website=force_website)
+
+    def write(self, vals):
+        if self.user_has_groups('base.group_portal'):
+            for fname, value in vals.items():
+                field = self._fields.get(fname)
+                if isinstance(field, fields.Many2one):
+                    model = getattr(self, fname)
+                    record = model.browse(value)
```

Comment:
Can't we access "comodel" on field and skip the "getattr", direclty doing self.env[model].browse(value) ?